### PR TITLE
dm-worker: fix Lightning version info

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/8618

### What is changed and how it works?
Use LDFLAGS to assign lightning's version info.
Because Lightning's version is assigned in an early time. https://github.com/pingcap/tidb/blob/e059bbdbb06780f4391ee01d8ee16f03fbaf41cb/br/pkg/lightning/backend/local/local.go#L115
Assign this in build function will not take effect. So we use LDFLAGS now.

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by lichunzhu** [comment]
Test is WIP. Reviewers can review the code first.

File: dm/loader/lightning.go [hunk 1]
```diff
@@ -83,8 +85,20 @@ type LightningLoader struct {
...
 }
 
+func initLightningVersionInfo() {
+	build.ReleaseVersion = version.ReleaseVersion
+	build.BuildTS = version.BuildTS
+	build.GitHash = version.GitHash
+	build.GitBranch = version.GitBranch
+}
+
+var syncLightningVersionOnce sync.Once
```
Comment (lance6716) [inline dm/loader/lightning.go:95]: as the PR description says, use init() can remove this variable.

**Review by lance6716 — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**